### ISSUE TITLE
set-constant.js scriptlet have stopped working (v1.25.3b2)

### ISSUE THREAD

**Issue by dimisa-RUAdList** [uBlockOrigin/uBlock-issues#947]
Google Chrome 80.0.3987.132

uBlock Origin
default + RU AdList + Counters (`https://easylist-downloads.adblockplus.org/cntblock.txt`)

Compare the work of versions 1.25.3b2 and 1.24.4 on sites:
```
https://www.rambler.ru/
https://trucks.ati.su/
https://news.mail.ru/economics/40959222/?frommail=1
https://it-actual.ru/securely-erase-hard-drive.html
https://video.khl.ru/quotes/695965
https://www.championat.com/boxing/_boxpro.html
```
Rules `set-constant.js, *, undefined` - behave very strange.

**Comment by gwarser** [comment]
Can you be more specific?

How this works:
	
	mail.ru##+js(set-constant, Object.prototype.ad/rectangle, undefined)
	mail.ru##+js(set-constant, Object.prototype.ad/tgb, undefined)

?

**Comment by ediowar** [comment]
Most likely, you need the Russian ip and the language of the browser.

**Comment by uBlock-user** [comment]
> behave very strange.

What does this mean ? Website can act strange depending on the value you set.

**Comment by uBlock-user** [comment]
Consider using `null` instead of `undefined` on such websites and see if you can still reproduce the strange behaviour you mentioned.

**Comment by ediowar** [comment]
It started after commit https://github.com/gorhill/uBlock/commit/14ebfbea279c3d949e270a03bab78e02e3e0dac4

**Comment by gorhill** [comment]
I can reproduce a difference on `https://video.khl.ru/quotes/695965`. I get the same behavior when I revert <https://github.com/gorhill/uBlock/commit/14ebfbea279c3d949e270a03bab78e02e3e0dac4#diff-d7616deaf3a39e3f19f8d70d098bfe7a>.

**Comment by gorhill** [comment]
So apparently `Object.prototype instanceof Object` is `false` -- I didn't expect this.

**Comment by dimisa-RUAdList** [comment]
1.25.3b2 vs 1.24.4 >

`https://www.rambler.ru/` - 
https://i.imgur.com/EQ06Rl9.jpg vs https://i.imgur.com/d1YWwmm.jpg
`rambler.*##+js(set-constant, Object.prototype.getPrefixes, undefined)`

`https://trucks.ati.su/` - 
https://i.imgur.com/L5BR8lZ.jpg vs https://i.imgur.com/SQ3V2oi.jpg
`ati.su##+js(set-constant, Object.prototype.checkBannerItem, undefined)`

`https://news.mail.ru/economics/40959222/?frommail=1` -
https://i.imgur.com/P3ChdNr.jpg vs https://i.imgur.com/FJORozn.jpg
`mail.ru##+js(set-constant, Object.prototype.mimic, undefined)`

Etc...

Using` nul`l instead of `undefined` does not solve the problem.

**Comment by gorhill** [comment]
Issue is fixed in 1.25.3b3.

**Comment by dimisa-RUAdList** [comment]
Now all is well!
